clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -179,7 +190,7 @@
  * would consume from last committed offset and would repeat the insert of the last batch of data. Used in this way
  * Kafka provides what is often called "at-least once delivery" guarantees, as each message will likely be delivered one
  * time but in failure cases could be duplicated.
```

Comment:
Does it make sense to mention somewhere that even with manual commits, offsets will be automatically committed in some situations (such as rebalancing)?